qualtran/bloqs/multi_control_multi_target_pauli_test.py

```diff
@@ -41,3 +45,39 @@ def test_t_complexity_mcp(num_controls: int, pauli: cirq.Pauli, cv: int):
     gate = MultiControlPauli([cv] * num_controls, target_gate=pauli)
     assert_valid_bloq_decomposition(gate)
     assert_decompose_is_consistent_with_t_complexity(gate)
+
+
+@pytest.mark.parametrize("cvs", [(), (0), (1, 0), (1, 1, 1), (1, 0, 1, 0)])
+def test_multi_control_x(cvs):
+    bloq = MultiControlX(cvs=cvs)
+    assert_valid_bloq_decomposition(bloq=bloq)
+
+
+@pytest.mark.parametrize(
+    "cvs,x,ctrl",
+    [((), 0, 0), ((0), 1, 0), ((1, 0), 0, 2), ((1, 1, 1), 1, 7), ((1, 0, 1, 0), 1, 10)],
+)
+def test_classical_multi_control_x(x, cvs, ctrl):
+    bloq = MultiControlX(cvs=cvs)
+    cbloq = bloq.decompose_bloq()
+    ret1 = bloq.call_classically(x=x, ctrl=ctrl)
+    ret2 = cbloq.call_classically(x=x, ctrl=ctrl)
+    assert ret1 == ret2
+
+
+@pytest.mark.parametrize(
+    "cvs,x,ctrl,result",
+    [
+        ((), 0, 0, 1),
+        ((0), 1, 0, 0),
+        ((1, 0), 0, 2, 1),
+        ((1, 1, 1), 1, 7, 0),
+        ((1, 0, 1, 0), 1, 10, 0),
+        ((1), 0, 0, 0),
+    ],
+)
+def test_classical_multi_control_x_correctness(x, cvs, ctrl, result):
+    bloq = MultiControlX(cvs=cvs)
+    cbloq = bloq.decompose_bloq()
+    ret = bloq.call_classically(x=x, ctrl=ctrl)
+    assert ret[-1] == result
```

Comment:
what is this testing? `cbloq` is unused

Reply:
I merged the two tests, but basically one will test that the output of the classical composite_bloq matches the classical implementation. The other test asserts that the value that matches both is actually correct.